lib/Dialects/LinalgTransform/Transforms/TransformInterpreter.cpp

```diff
@@ -149,80 +144,51 @@ executePadFromTileOp(linalg::LinalgOp target, linalg::transform::TileOp tileOp,
   paddingOptions.setPaddingNoFoldComputationFunction(packFunc);
   paddingOptions.setPaddingHoistComputationFunction(hoistingFunc);
 
-  MLIRContext *ctx = target->getContext();
-  FuncOp parentFunc = target->getParentOfType<FuncOp>();
-  RewritePatternSet patterns(ctx);
-  patterns.insert<LinalgPaddingPattern>(ctx, paddingOptions,
-                                        makeTransformationFilter(target));
-  if (failed(applyPatternsAndFoldGreedily(target->getParentOfType<FuncOp>(),
-                                          std::move(patterns))))
-    return failure();
-
-  findMarkedOps(parentFunc, nextTargets);
-  return success();
+  return callFunctional<LinalgPaddingPattern>(tileOp.getContext(),
+                                              paddingOptions);
 }
 
 /// Applies the generalization pattern to the given target operation as
 /// indicated by the tile op that subsumes padding. Populates `nextTargets` with
 /// transformable operations for further transformations (currently, the single
 /// generalized op).
-static LogicalResult
-executeGeneralizeFromTileOp(linalg::LinalgOp target,
-                            linalg::transform::TileOp tileOp,
-                            SmallVectorImpl<Operation *> &nextTargets) {
-  if (!target)
-    return failure();
-
-  if (!tileOp.generalize()) {
-    nextTargets.push_back(target);
-    return success();
-  }
-
-  MLIRContext *ctx = target->getContext();
-  FuncOp parentFunc = target->getParentOfType<FuncOp>();
-  RewritePatternSet patterns(ctx);
-  patterns.insert<LinalgGeneralizationPattern>(
-      ctx, makeTransformationFilter(target));
-  if (failed(applyPatternsAndFoldGreedily(target->getParentOfType<FuncOp>(),
-                                          std::move(patterns))))
-    return failure();
-
-  findMarkedOps(parentFunc, nextTargets);
-  return success();
+static FunctionalLinalgTransform
+buildGeneralizeFromTileOpPattern(linalg::transform::TileOp tileOp) {
+  if (!tileOp.generalize())
+    return forwardOp;
+  auto pattern =
+      callFunctional<LinalgGeneralizationPattern>(tileOp.getContext());
+  return [pattern = std::move(pattern)](
+             LinalgOp op, PatternRewriter &rewriter) -> FailureOr<LinalgOp> {
+    FailureOr<GenericOp> result = pattern(op, rewriter);
```

Comment:
Feel free to make LinalgGeneralizationPattern::returningMatchAndRewrite return FailureOr<GenericOp> upstream if  you think it is better.

Reply:
`callFunctional`  does invoke `returningMatchAndRewrite` but because some of these return `GenericOp` instead of `LinalgOp`, I have to wrap them again to cast to `LinalgOp`